From d966c9cd35db0fa54b507dd1f346a5647fceea0b Mon Sep 17 00:00:00 2001
From: Ahmad Samir <a.samirh78@gmail.com>
Date: Wed, 8 Oct 2025 01:05:45 +0300
Subject: [PATCH 359/553] tst_QPlainTextEdit: port away from QPair

Pick-to: 6.8 6.5
Task-number: QTBUG-115841
Change-Id: I24806fd4e01ca7d74983b6a914aa68252458b508
Reviewed-by: Ivan Solovev <ivan.solovev@qt.io>
(cherry picked from commit 4346e5523f41ebe4a023b4a112dc713ec43d2279)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 .../qplaintextedit/tst_qplaintextedit.cpp     | 65 +++++++++++--------
 1 file changed, 37 insertions(+), 28 deletions(-)

diff --git a/tests/auto/widgets/widgets/qplaintextedit/tst_qplaintextedit.cpp b/tests/auto/widgets/widgets/qplaintextedit/tst_qplaintextedit.cpp
index ca7cc6d4b46..f5b1cc00c9e 100644
--- a/tests/auto/widgets/widgets/qplaintextedit/tst_qplaintextedit.cpp
+++ b/tests/auto/widgets/widgets/qplaintextedit/tst_qplaintextedit.cpp
@@ -25,7 +25,7 @@
 #include "../../../shared/platformclipboard.h"
 
 //Used in copyAvailable
-typedef QPair<Qt::Key, Qt::KeyboardModifier> keyPairType;
+using keyPairType = std::pair<Qt::Key, Qt::KeyboardModifier>;
 typedef QList<keyPairType> pairListType;
 Q_DECLARE_METATYPE(keyPairType);
 
@@ -923,71 +923,80 @@ void tst_QPlainTextEdit::copyAvailable_data()
     QTest::addColumn<QList<bool> >("copyAvailable");
     QTest::addColumn<QString>("function");
 
-    pairListType keystrokes;
     QList<bool> copyAvailable;
 
-    keystrokes << qMakePair(Qt::Key_B, Qt::NoModifier) <<  qMakePair(Qt::Key_B, Qt::NoModifier)
-               << qMakePair(Qt::Key_Left, Qt::ShiftModifier);
+    pairListType keystrokes = {{Qt::Key_B, Qt::NoModifier},
+                               {Qt::Key_B, Qt::NoModifier},
+                               {Qt::Key_Left, Qt::ShiftModifier}};
     copyAvailable << true ;
     QTest::newRow(QString("Case1 B,B, <- + shift | signals: true").toLatin1())
         << keystrokes << copyAvailable << QString();
 
-    keystrokes.clear();
     copyAvailable.clear();
 
-    keystrokes << qMakePair(Qt::Key_T, Qt::NoModifier) << qMakePair(Qt::Key_A, Qt::NoModifier)
-               <<  qMakePair(Qt::Key_A, Qt::NoModifier) << qMakePair(Qt::Key_Left, Qt::ShiftModifier);
+    keystrokes = {{Qt::Key_T, Qt::NoModifier},
+                  {Qt::Key_A, Qt::NoModifier},
+                  {Qt::Key_A, Qt::NoModifier},
+                  {Qt::Key_Left, Qt::ShiftModifier}};
     copyAvailable << true << false;
     QTest::newRow(QString("Case2 T,A,A, <- + shift, cut() | signals: true, false").toLatin1())
         << keystrokes << copyAvailable << QString("cut");
 
-    keystrokes.clear();
     copyAvailable.clear();
 
-    keystrokes << qMakePair(Qt::Key_T, Qt::NoModifier) << qMakePair(Qt::Key_A, Qt::NoModifier)
-               <<  qMakePair(Qt::Key_A, Qt::NoModifier) << qMakePair(Qt::Key_Left, Qt::ShiftModifier)
-               << qMakePair(Qt::Key_Left, Qt::ShiftModifier)  << qMakePair(Qt::Key_Left, Qt::ShiftModifier);
+    keystrokes = {{Qt::Key_T, Qt::NoModifier},
+                  {Qt::Key_A, Qt::NoModifier},
+                  {Qt::Key_A, Qt::NoModifier},
+                  {Qt::Key_Left, Qt::ShiftModifier},
+                  {Qt::Key_Left, Qt::ShiftModifier},
+                  {Qt::Key_Left, Qt::ShiftModifier}};
     copyAvailable << true;
     QTest::newRow(QString("Case3 T,A,A, <- + shift, <- + shift, <- + shift, copy() | signals: true").toLatin1())
         << keystrokes << copyAvailable << QString("copy");
 
-    keystrokes.clear();
     copyAvailable.clear();
 
-    keystrokes << qMakePair(Qt::Key_T, Qt::NoModifier) << qMakePair(Qt::Key_A, Qt::NoModifier)
-               <<  qMakePair(Qt::Key_A, Qt::NoModifier) << qMakePair(Qt::Key_Left, Qt::ShiftModifier)
-               << qMakePair(Qt::Key_Left, Qt::ShiftModifier)  << qMakePair(Qt::Key_Left, Qt::ShiftModifier)
-               << qMakePair(Qt::Key_X, Qt::ControlModifier);
+    keystrokes = {{Qt::Key_T, Qt::NoModifier},
+                  {Qt::Key_A, Qt::NoModifier},
+                  {Qt::Key_A, Qt::NoModifier},
+                  {Qt::Key_Left, Qt::ShiftModifier},
+                  {Qt::Key_Left, Qt::ShiftModifier},
+                  {Qt::Key_Left, Qt::ShiftModifier},
+                  {Qt::Key_X, Qt::ControlModifier}};
     copyAvailable << true << false;
     QTest::newRow(QString("Case4 T,A,A, <- + shift, <- + shift, <- + shift, ctrl + x, paste() | signals: true, false").toLatin1())
         << keystrokes << copyAvailable << QString("paste");
 
-    keystrokes.clear();
     copyAvailable.clear();
 
-    keystrokes << qMakePair(Qt::Key_B, Qt::NoModifier) <<  qMakePair(Qt::Key_B, Qt::NoModifier)
-               << qMakePair(Qt::Key_Left, Qt::ShiftModifier) << qMakePair(Qt::Key_Left, Qt::NoModifier);
+    keystrokes = {{Qt::Key_B, Qt::NoModifier},
+                  {Qt::Key_B, Qt::NoModifier},
+                  {Qt::Key_Left, Qt::ShiftModifier},
+                  {Qt::Key_Left, Qt::NoModifier}};
     copyAvailable << true << false;
     QTest::newRow(QString("Case5 B,B, <- + shift, <- | signals: true, false").toLatin1())
         << keystrokes << copyAvailable << QString();
 
-    keystrokes.clear();
     copyAvailable.clear();
 
-    keystrokes << qMakePair(Qt::Key_B, Qt::NoModifier) <<  qMakePair(Qt::Key_A, Qt::NoModifier)
-               << qMakePair(Qt::Key_Left, Qt::ShiftModifier) << qMakePair(Qt::Key_Left, Qt::NoModifier)
-               << qMakePair(Qt::Key_Right, Qt::ShiftModifier);
+    keystrokes = {{Qt::Key_B, Qt::NoModifier},
+                  {Qt::Key_A, Qt::NoModifier},
+                  {Qt::Key_Left, Qt::ShiftModifier},
+                  {Qt::Key_Left, Qt::NoModifier},
+                  {Qt::Key_Right, Qt::ShiftModifier}};
     copyAvailable << true << false << true << false;
     QTest::newRow(QString("Case6 B,A, <- + shift, ->, <- + shift | signals: true, false, true, false").toLatin1())
         << keystrokes << copyAvailable << QString("cut");
 
-    keystrokes.clear();
     copyAvailable.clear();
 
-    keystrokes << qMakePair(Qt::Key_T, Qt::NoModifier) << qMakePair(Qt::Key_A, Qt::NoModifier)
-               <<  qMakePair(Qt::Key_A, Qt::NoModifier) << qMakePair(Qt::Key_Left, Qt::ShiftModifier)
-               << qMakePair(Qt::Key_Left, Qt::ShiftModifier)  << qMakePair(Qt::Key_Left, Qt::ShiftModifier)
-               << qMakePair(Qt::Key_X, Qt::ControlModifier);
+    keystrokes = {{Qt::Key_T, Qt::NoModifier},
+                  {Qt::Key_A, Qt::NoModifier},
+                  {Qt::Key_A, Qt::NoModifier},
+                  {Qt::Key_Left, Qt::ShiftModifier},
+                  {Qt::Key_Left, Qt::ShiftModifier},
+                  {Qt::Key_Left, Qt::ShiftModifier},
+                  {Qt::Key_X, Qt::ControlModifier}};
     copyAvailable << true << false << true;
     QTest::newRow(QString("Case7 T,A,A, <- + shift, <- + shift, <- + shift, ctrl + x, undo() | signals: true, false, true").toLatin1())
         << keystrokes << copyAvailable << QString("undo");
-- 
2.51.2

